Migrated  article explaining how to connect to Kafka with Java (#901)
aiven/devportal: docs/products/kafka/howto/connect-java.rst

> Java samples for testing Aiven for Apache Kafka®

Reviewer: Can we use the standard "Connect with Java" title?

=================================================

> The sample Java code in this article demonstrates how to connect to your Aiven for Apache Kafka service and send messages into a topic.

Reviewer: We're not sending messages in a topic anymore :D 
```suggestion
The sample Java code in this article demonstrates how to connect to your Aiven for Apache Kafka service as producer or consumer of messages.
```

Author: I removed that part completely :)

.. note::

    The examples in this article provide two different options for authentication: SSL and SASL-SSL. For more information on these  authentication methods read :doc:`our article on Kafka authentication types <../concepts/auth-types>`.

.. note::

    The instructions below assume that there is existing topic with name ``demo-topic`` and that you have created the keystore ``client.keystore.p12`` and truststore ``client.truststore.jks``. For details, see :doc:`our article on configuring Java SSL to access Kafka <../howto/keystore-truststore>`.

> Relying on SSL authentication

Reviewer: Can we add a "Variables" section where we include all the possible variables (including the folder where we store the keystore/truststore)?

Author: done!

> Relying on SSL authentication

Reviewer: I would also change the title to "With SSL authentication" (not using the `ing` in the title)

Author: done

-----------------------------

> Create and run a producer, that sends a message into the topic ``demo-topic``:

Reviewer: Should we have subtitles for Producer and Consumer? Also worth checking with @TibsAtWork since for python we only give instructions on how to connecT?

Author: Aligned with the python example and focused on connect part

.. code::

         package io.aiven.kafkaExample;
         import java.io.IOException;
         import java.util.Properties;
         import org.apache.kafka.clients.producer.KafkaProducer;
         import org.apache.kafka.clients.producer.ProducerRecord;

         public class Producer {
             public static void main(String[] args) throws IOException {
                 Properties props = new Properties();

>                  props.put("bootstrap.servers", "service-uri.aivencloud.com:port");

Reviewer: Can we reference variables instead of strings?

Author: done!

                 props.put("security.protocol", "SSL");
                 props.put("ssl.truststore.location", "client.truststore.jks");
                 props.put("ssl.truststore.password", "secret");
                 props.put("ssl.keystore.type", "PKCS12");
                 props.put("ssl.keystore.location", "client.keystore.p12");
                 props.put("ssl.keystore.password", "secret");
                 props.put("ssl.key.password", "secret");
                 props.put("key.serializer",

>                      "org.apache.kafka.common.serialization.StringSerializer");

Reviewer: I believe serializers need to be explained a bit? Maybe just saying what the options are?
This becomes quite a complex topic with Avro

Author: I linked to the list from Apache Kafka documentation. Do you think it is enough, or should I touch the topic of Avro?

                 props.put("ssl.keystore.location", "client.keystore.p12");
                 props.put("ssl.keystore.password", "secret");
                 props.put("ssl.key.password", "secret");
                 props.put("group.id", "demo-group");
                 props.put("key.deserializer",
                   "org.apache.kafka.common.serialization.StringDeserializer");
                 props.put("value.deserializer",
                   "org.apache.kafka.common.serialization.StringDeserializer");
                 KafkaConsumer<String, String> consumer = new KafkaConsumer<>(props);
                 consumer.subscribe(Arrays.asList("demo-topic"));
                 while (true) {
                     ConsumerRecords<String, String> records = consumer.poll(1000);
                     for (ConsumerRecord<String, String> record : records) {
                         System.out.printf("offset = %d, key = %s, value = %s",
                             record.offset(), record.key(), record.value());
                     }
                 }
             }
         }

Relying on SASL-SSL authentication
-----------------------------------

> If you want to use SASL authentication, turn on ``kafka_authentication_methods.sasl`` under *Advanced configuration* on the *Overview* page of your service.

Reviewer: We should have an article to reference here https://developer.aiven.io/docs/products/kafka/howto/kafka-sasl-auth.html

Author: done
